apps/molgenis-components/src/components/forms/EditModal.vue

```diff
@@ -0,0 +1,231 @@
+    <template #body>
+      <RowEdit
+        :id="id"
+        v-model="rowData"
+        :pkey="pkey"
+        :tableName="tableName"
+        :tableMetaData="tableMetaData"
+        :graphqlURL="graphqlURL"
+        :clone="clone"
+      >
+      </RowEdit>
+    </template>
+    <template #footer>
+      <RowEditFooter
+        class="modal-footer"
+        :id="id + '-footer'"
+        :tableName="tableName"
+        :errorMessage="errorMessage"
+        @cancel="handleClose"
+        @saveDraft="handleSaveDraftRequest"
+        @save="handleSaveRequest"
+      ></RowEditFooter>
+    </template>
+  </LayoutModal>
+</template>
+
+<script>
+import Client from "../../client/client.js";
+import LayoutModal from "../layout/LayoutModal.vue";
+import RowEditFooter from "./RowEditFooter.vue";
+import RowEdit from "./RowEdit.vue";
+import { filterObject } from "../utils";
+
+export default {
+  name: "EditModal",
+  components: { LayoutModal, RowEditFooter, RowEdit },
+  data() {
+    return {
+      rowData: {},
+      tableMetaData: {},
+      client: null,
+      errorMessage: null,
+    };
+  },
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+    tableName: {
+      type: String,
+      required: true,
+    },
+    isModalShown: {
+      type: Boolean,
+      required: true,
+    },
+    graphqlURL: {
+      type: String,
+      required: false,
+      default: () => "graphql",
+    },
+    pkey: {
+      type: Object,
+      required: false,
+      default: () => null,
+    },
+    clone: {
+      type: Boolean,
+      required: false,
+      default: () => false,
+    },
+    visibleColumns: {
+      type: Array,
+      required: false,
+      default: () => [],
+    },
+    defaultValue: {
+      type: Object,
+      required: false,
+      default: () => null,
+    },
+  },
+  computed: {
+    title() {
+      return this.titlePrefix + " " + this.tableName;
+    },
+    titlePrefix() {
+      return this.pkey && this.clone ? "copy" : this.pkey ? "update" : "insert";
+    },
+  },
+  methods: {
+    handleSaveRequest() {
+      this.save({ ...this.rowData, mg_draft: false });
+    },
+    handleSaveDraftRequest() {
+      this.save({ ...this.rowData, mg_draft: true });
+    },
+    async save(formData) {
+      this.errorMessage = null;
+      const action =
+        this.pkey && !this.clone ? "updateDataRow" : "insertDataRow";
+      const result = await this.client[action](
+        formData,
+        this.tableName,
+        this.graphqlURL
+      ).catch(this.handleSaveError);
+      if (result) {
+        this.$emit("close");
+      }
+    },
+    handleSaveError(error) {
+      this.errorMessage =
+        error.response?.status === 403
+          ? "Schema doesn't exist or permission denied. Do you need to Sign In?"
+          : error.response
+          ? error.response.data.errors[0].message
+          : "An Error occurred during save";
```

Comment:
If you start nesting ternaries, it is better for readability to switch to if else